### PR TITLE
Improve smartyClassname()

### PR DESCRIPTION
**Pull request by justeen35**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Improve smartyClassname() . It's always a good idea to validate function input, to prevent unexpected behavior.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Green CI
| Fixed ticket?     | 
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

### CODE REVIEW

File: config/smarty.config.inc.php [hunk 1]
```diff
@@ -177,6 +177,10 @@ function smartyCleanHtml($data)
 
 function smartyClassname($classname)
 {
+    if (!is_string($classname)) {
```
Comment (FabienPapet) [inline config/smarty.config.inc.php]: Why not adding a `string` type to the parameter ?

File: config/smarty.config.inc.php [hunk 2]
```diff
@@ -177,6 +177,10 @@ function smartyCleanHtml($data)
...
+        throw new InvalidArgumentException('The $classname parameter must be a string.');
+    }
+
```
Comment (FabienPapet) [inline config/smarty.config.inc.php]:
```suggestion
function smartyClassname(string $classname)
{
    
```

**Review by FabienPapet — CHANGES_REQUESTED** [review]

File: config/smarty.config.inc.php (hunk 1 above)
Comment (justeen35) [reply config/smarty.config.inc.php]: Wooups I thought you would tell me "no it's a BC break" 🤣

File: config/smarty.config.inc.php [hunk 3]
```diff
@@ -175,8 +175,9 @@ function smartyCleanHtml($data)
     }
 }
 
-function smartyClassname($classname)
+function smartyClassname(string $classname)
 {
+
```
Comment (kpodemski) [inline config/smarty.config.inc.php]:
```suggestion
```
no need for this empty line 👍🏻 other than that, lgtm 👍🏻

Comment (justeen35) [reply config/smarty.config.inc.php]: Done

**Review by kpodemski — APPROVED** [review]

**Review by FabienPapet — APPROVED** [review]

**Comment by kpodemski** [comment]
thank you, @justeen35 :)

**Pull request merged**